AI/BattleAI/StackWithBonuses.cpp

```diff
@@ -18,7 +18,8 @@ void actualizeEffect(TBonusListPtr target, const Bonus & ef)
 	{
 		if(bonus->source == Bonus::SPELL_EFFECT && bonus->type == ef.type && bonus->subtype == ef.subtype)
 		{
-			bonus->turnsRemain = std::max(bonus->turnsRemain, ef.turnsRemain);
+			//TODO: this code changes gamestate
+			//bonus->turnsRemain = std::max(bonus->turnsRemain, ef.turnsRemain);
```

Comment:
And here is what causes this issue. Bonuses from TBonusListPtr target can be from real stacks from gamestate.

Reply:
After  commenting out this code I was not able to reproduce the crash again.

Reply:
Cloned the bonus from the gamestate and changed the clone